### PR TITLE
Trello

### PR DESCRIPTION
**Pull request by adambaumeister**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Description
This introduced a new pack for managing Trello content.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [X] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

### CODE REVIEW

File: Packs/Trello/Integrations/Trello/Trello.yml [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+      isArray: false
+      name: board_id
+      required: false
+      secret: false
+    - auto: PREDEFINED
+      default: false
+      description: CSV of action types used to filter response.
+      isArray: false
+      name: filter
+      predefined:
```
Comment (Itay4) [inline Packs/Trello/Integrations/Trello/Trello.yml:197]: Something seems off with the predefined values here:
1. the first one is an empty value 
2. the rest have `|\n` - is it intentional ?

File: Packs/Trello/Integrations/Trello/action_types.csv [hunk 2]
```diff
@@ -0,0 +1,67 @@
+,acceptEnterpriseJoinRequest
```
Comment (Itay4) [inline Packs/Trello/Integrations/Trello/action_types.csv:1]: What is the purpose of this file?

File: Packs/Trello/Integrations/Trello/Trello.py [hunk 3]
```diff
@@ -0,0 +1,683 @@
...
+
+
+def get_remote_data(client, args):
+    parsed_args = GetRemoteDataArgs(args)
+    new_incident_data = client.get_card(parsed_args.remote_incident_id)
+    demisto.info(f"Calling Remote data fetch for {parsed_args.remote_incident_id}")
+
+    new_incident_data['dbotMirrorInstance'] = demisto.integrationInstance()
+
+    # demisto.info(json.dumps(new_incident_data, indent=4, sort_keys=True))
```
Comment (Itay4) [inline Packs/Trello/Integrations/Trello/Trello.py:572]: can we remove the comment?

File: Packs/Trello/Integrations/Trello/Trello.yml [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+    - default: false
+      description: Last update
+      isArray: false
+      name: lastUpdate
+      required: false
+      secret: false
+    deprecated: false
+    description: Used by mirroring
+    execution: false
+    name: get-remote-data
```
Comment (Itay4) [inline Packs/Trello/Integrations/Trello/Trello.yml:387]: did you plan to implement incoming mirroring? 
if so, some stuff are missing, such as the `isremotesyncin: true` is missing from the YML and the handling in the code and the mapper

File: Packs/Trello/pack_metadata.json [hunk 5]
```diff
@@ -0,0 +1,14 @@
+{
+    "name": "Trello",
+    "description": "Trello is a card-based activity tracker.",
+    "support": "community",
+    "currentVersion": "1.0.0",
+    "author": "Adam Baumeister",
+    "url": "https://www.paloaltonetworks.com/cortex",
```
Comment (Itay4) [inline Packs/Trello/pack_metadata.json:7]: I'm going to modify the pack metadata a bit (remove the url, add category, tags, usecases and so on - feel free to add them yourself if you wish)

**Review by Itay4 — COMMENTED** [review]
Thanks for the contribution, overall looks good 👍 
A few general comments:
1. Please modify all commands names to have the prefix `trello-`
2. Please make sure all context outputs have the types set in the YML file
3. The integration and the playbooks has `fromversion: 6.0.0` and the rest of the entities have `fromversion: 5.0.0` - I think we should go all to be `fromversion: 6.0.0`. If you agree, I can help with doing that.
4. Did you consider adding a classifier as well?

File: Packs/Trello/Integrations/Trello/Trello.yml (hunk 4 above)
Comment (Itay4) [reply Packs/Trello/Integrations/Trello/Trello.yml:387]: As discussed, this command should be removed from the YML file.

File: Packs/Trello/Integrations/Trello/Trello.yml [hunk 6]
```diff
@@ -0,0 +1,520 @@
...
+    for all integration commands.
+  display: Default and Fetch Board ID
+  hidden: false
+  name: board_id
+  required: false
+  type: 0
+- display: API Key
+  hidden: false
+  name: credentials
+  required: false
```
Comment (Itay4) [inline Packs/Trello/Integrations/Trello/Trello.yml:29]:
```suggestion
  required: true
```

File: Packs/Trello/Classifiers/classifier-Trello_-_Incoming_Mapper.json [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+					"complex": null,
+					"simple": "create_action.board_id"
+				},
+				"Trello Board Name": {
+					"complex": null,
+					"simple": "create_action.list_name"
+				},
+				"Trello Card ID": {
+					"complex": null,
+					"simple": "ID"
```
Comment (Itay4) [inline Packs/Trello/Classifiers/classifier-Trello_-_Incoming_Mapper.json:29]: Shouldn't that be `id` instead of `ID` ?

**Review by Itay4 — COMMENTED** [review]
Following the demo, added a few more comments

File: Packs/Trello/pack_metadata.json [hunk 8]
```diff
@@ -0,0 +1,14 @@
+{
+    "name": "Trello",
+    "description": "Trello is a card-based activity tracker.",
```
Comment (sangsoojun88) [inline Packs/Trello/pack_metadata.json:3]: -> Trello is a card-based activity tracker. Utilize this content pack to organize and prioritize your personal and work life using boards, lists, cards, and so on.

**Comment by adambaumeister** [comment]
Hiya, 

I've been on training all week this week, I'll start working on this again now.

Cheers
Adam

File: Packs/Trello/Integrations/Trello/action_types.csv (hunk 2 above)
Comment (adambaumeister) [reply Packs/Trello/Integrations/Trello/action_types.csv:1]: This file can be removed from the PR if preferred. It contains teh list of every possible card action, which we use to generate the update-card options.

File: Packs/Trello/Integrations/Trello/Trello.yml (hunk 4 above)
Comment (adambaumeister) [reply Packs/Trello/Integrations/Trello/Trello.yml]: Removed

File: Packs/Trello/IncidentFields/incidentfield-Trello_Board_ID.json [hunk 9]
```diff
@@ -0,0 +1,30 @@
+{
+    "associatedToAll": true,
```
[the remark this answers is not in the document]
Comment (adambaumeister) [reply Packs/Trello/IncidentFields/incidentfield-Trello_Board_ID.json]: Done

File: Packs/Trello/IncidentFields/incidentfield-Trello_Board_Name.json [hunk 10]
```diff
@@ -0,0 +1,32 @@
+{
+    "associatedToAll": false,
+    "associatedTypes": [
+        "Trello"
```
[the remark this answers is not in the document]
Comment (adambaumeister) [reply Packs/Trello/IncidentFields/incidentfield-Trello_Board_Name.json:4]: Adedd

File: Packs/Trello/Integrations/Trello/Trello.yml (hunk 1 above)
Comment (adambaumeister) [reply Packs/Trello/Integrations/Trello/Trello.yml:197]: Fixed - came across incorrectly

**Comment by adambaumeister** [comment]
Added a large amount of fixes based on feedback. Re-ran test playbook, confirmed still working.

File: Packs/Trello/IncidentTypes/Trello.json [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+		"default": false,
+		"autorun": false,
+		"preProcessingScript": "",
+		"closureScript": "",
+		"disabled": false,
+		"reputationCalc": 0,
+		"layout": ""
+	},
+	{
+		"id": "Trello",
```
Comment (Itay4) [inline Packs/Trello/IncidentTypes/Trello.json:135]: @adambaumeister There are a few types here, I suppose you want to keep only the last one right?

File: Packs/Trello/Integrations/Trello/README.md [hunk 12]
```diff
@@ -0,0 +1,639 @@
...
+
+There is no context output for this command.
+
+#### Command Example
+``` ```
+
+#### Human Readable Output
+
+
+
```
Comment (Itay4) [inline Packs/Trello/Integrations/Trello/README.md:384]:
```suggestion
```

File: Packs/Trello/Integrations/Trello/Trello.py [hunk 13]
```diff
@@ -0,0 +1,682 @@
...
+        elif demisto.command() == 'trello-add-comment':
+            card_id = demisto.args().get("card_id")
+            return_results(add_comment(client, card_id, demisto.args()))
+        elif demisto.command() == 'trello-delete-card':
+            card_id = demisto.args().get("card_id")
+            return_results(delete_card(client, card_id))
+        elif demisto.command() == 'trello-get-mapping-fields':
+            demisto.results(get_mapping_fields())
+        elif demisto.command() == 'trello-get-remote-data':
+            demisto.results(get_remote_data(client, demisto.args()))
```
Comment (Itay4) [inline Packs/Trello/Integrations/Trello/Trello.py:674]: do you wish to keep those in the code for the future?

File: Packs/Trello/IncidentFields/incidentfield-Trello_Board_ID.json [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+    "required": false,
+    "sla": 0,
+    "system": false,
+    "threshold": 72,
+    "type": "shortText",
+    "unmapped": false,
+    "unsearchable": false,
+    "useAsKpi": false,
+    "version": -1,
+    "fromVersion": "5.0.0"
```
Comment (Itay4) [inline Packs/Trello/IncidentFields/incidentfield-Trello_Board_ID.json:32]:
```suggestion
    "fromVersion": "6.0.0"
```

File: Packs/Trello/Classifiers/classifier-Trello_-_Incoming_Mapper.json [hunk 15]
```diff
@@ -0,0 +1,87 @@
+{
+	"brands": null,
+	"defaultIncidentType": "",
+	"description": "",
+	"feed": false,
+	"fromServerVersion": "",
+	"id": "a1be1481-cc66-4b81-8491-6fb6e9a7b9e5",
```
Comment (Itay4) [inline Packs/Trello/Classifiers/classifier-Trello_-_Incoming_Mapper.json]:
```suggestion
	"id": "Trello - Incoming Mapper",
```

File: Packs/Trello/Classifiers/classifier-Trello_-_Incoming_Mapper.json [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+	"sourceClassifierId": "",
+	"system": false,
+	"toServerVersion": "",
+	"transformer": {
+		"complex": null,
+		"simple": ""
+	},
+	"type": "mapping-incoming",
+	"unclassifiedCases": null,
+	"version": -1
```
Comment (Itay4) [inline Packs/Trello/Classifiers/classifier-Trello_-_Incoming_Mapper.json]:
```suggestion
	"version": -1,
	"fromVersion": "6.0.0"
```

File: Packs/Trello/IncidentFields/incidentfield-Trello_Card_ID.json [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+    "required": false,
+    "sla": 0,
+    "system": false,
+    "threshold": 72,
+    "type": "shortText",
+    "unmapped": false,
+    "unsearchable": false,
+    "useAsKpi": false,
+    "version": -1,
+    "fromVersion": "5.0.0"
```
Comment (Itay4) [inline Packs/Trello/IncidentFields/incidentfield-Trello_Card_ID.json:32]:
```suggestion
    "fromVersion": "6.0.0"
```

File: Packs/Trello/IncidentFields/incidentfield-Trello_Card_URL.json [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+    "required": false,
+    "sla": 0,
+    "system": false,
+    "threshold": 72,
+    "type": "url",
+    "unmapped": false,
+    "unsearchable": false,
+    "useAsKpi": false,
+    "version": -1,
+    "fromVersion": "5.0.0"
```
Comment (Itay4) [inline Packs/Trello/IncidentFields/incidentfield-Trello_Card_URL.json:32]:
```suggestion
    "fromVersion": "6.0.0"
```

File: Packs/Trello/IncidentFields/incidentfield-Trello_Created_By.json [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+    "required": false,
+    "sla": 0,
+    "system": false,
+    "threshold": 72,
+    "type": "shortText",
+    "unmapped": false,
+    "unsearchable": false,
+    "useAsKpi": false,
+    "version": -1,
+    "fromVersion": "5.0.0"
```
Comment (Itay4) [inline Packs/Trello/IncidentFields/incidentfield-Trello_Created_By.json:32]:
```suggestion
    "fromVersion": "6.0.0"
```

File: Packs/Trello/IncidentFields/incidentfield-Trello_Details.json [hunk 20]
```diff
@@ -0,0 +1,33 @@
...
+    "required": false,
+    "sla": 0,
+    "system": false,
+    "threshold": 72,
+    "type": "markdown",
+    "unmapped": false,
+    "unsearchable": false,
+    "useAsKpi": false,
+    "version": -1,
+    "fromVersion": "5.0.0"
```
Comment (Itay4) [inline Packs/Trello/IncidentFields/incidentfield-Trello_Details.json:32]:
```suggestion
    "fromVersion": "6.0.0"
```

File: Packs/Trello/IncidentFields/incidentfield-Trello_List_ID.json [hunk 21]
```diff
@@ -0,0 +1,33 @@
...
+    "required": false,
+    "sla": 0,
+    "system": false,
+    "threshold": 72,
+    "type": "shortText",
+    "unmapped": false,
+    "unsearchable": false,
+    "useAsKpi": false,
+    "version": -1,
+    "fromVersion": "5.0.0"
```
Comment (Itay4) [inline Packs/Trello/IncidentFields/incidentfield-Trello_List_ID.json:32]:
```suggestion
    "fromVersion": "6.0.0"
```

File: Packs/Trello/IncidentFields/incidentfield-Trello_List_Name.json [hunk 22]
```diff
@@ -0,0 +1,33 @@
...
+    "required": false,
+    "sla": 0,
+    "system": false,
+    "threshold": 72,
+    "type": "shortText",
+    "unmapped": false,
+    "unsearchable": false,
+    "useAsKpi": false,
+    "version": -1,
+    "fromVersion": "5.0.0"
```
Comment (Itay4) [inline Packs/Trello/IncidentFields/incidentfield-Trello_List_Name.json:32]:
```suggestion
    "fromVersion": "6.0.0"
```

File: Packs/Trello/IncidentFields/incidentfield-Trello_SLA.json [hunk 23]
```diff
@@ -0,0 +1,33 @@
...
+    "required": false,
+    "sla": 0,
+    "system": false,
+    "threshold": 0,
+    "type": "timer",
+    "unmapped": false,
+    "unsearchable": false,
+    "useAsKpi": false,
+    "version": -1,
+    "fromVersion": "5.0.0"
```
Comment (Itay4) [inline Packs/Trello/IncidentFields/incidentfield-Trello_SLA.json:32]:
```suggestion
    "fromVersion": "6.0.0"
```

File: Packs/Trello/Integrations/Trello/action_types.csv (hunk 2 above)
Comment (Itay4) [reply Packs/Trello/Integrations/Trello/action_types.csv:1]: ok so please lets either remove it from the PR or move it to a directory named `test_data` inside the `Packs/Trello/Integrations/Trello` directory

File: Packs/Trello/pack_metadata.json [hunk 24]
```diff
@@ -0,0 +1,14 @@
+{
+    "name": "Trello",
+    "description": "Trello is a card-based activity tracker. Use this content pack to organize and prioritize your personal and work life using boards, lists, cards and so on.",
+    "support": "community",
+    "currentVersion": "1.0.0",
+    "author": "Adam Baumeister",
+    "url": "https://www.paloaltonetworks.com/cortex",
```
Comment (Itay4) [inline Packs/Trello/pack_metadata.json:7]:
```suggestion
    "url": "",
```

File: Packs/Trello/pack_metadata.json [hunk 25]
```diff
@@ -0,0 +1,14 @@
...
+    "email": "",
+    "created": "2021-03-29T07:34:09Z",
+    "categories": [],
```
Comment (Itay4) [inline Packs/Trello/pack_metadata.json:10]:
```suggestion
    "categories": [
        "Case Management"
    ],
```

File: Packs/Trello/Playbooks/playbook-Trello_-_Generic.yml [hunk 26]
```diff
@@ -0,0 +1,431 @@
...
+        "width": 780,
+        "x": 250,
+        "y": 50
+      }
+    }
+  }
+tests:
+- c98f69a4-1154-49c1-88b8-40330fb8d6c5
+fromversion: 6.0.0
+description: Input LIST
```
Comment (Itay4) [inline Packs/Trello/Playbooks/playbook-Trello_-_Generic.yml:431]: Can we improve the description here a bit?

File: Packs/Trello/Playbooks/playbook-Trello_-_Generic.yml [hunk 27]
```diff
@@ -0,0 +1,431 @@
...
+      "dimensions": {
+        "height": 1605,
+        "width": 780,
+        "x": 250,
+        "y": 50
+      }
+    }
+  }
+tests:
+- c98f69a4-1154-49c1-88b8-40330fb8d6c5
```
Comment (Itay4) [inline Packs/Trello/Playbooks/playbook-Trello_-_Generic.yml:429]:
```suggestion
- No tests
```

File: Packs/Trello/Integrations/Trello/Trello.yml [hunk 28]
```diff
@@ -0,0 +1,436 @@
...
+    - contextPath: Trello.Cards.Desc
+      description: Card description
+      type: Unknown
+    - contextPath: Trello.Cards.Start
+      description: Card Start date
+      type: Date
+    - contextPath: Trello.Cards.Labels.id
+      description: Label ID
+      type: Unknown
+  dockerimage: demisto/python3:3.9.2.18414
```
Comment (Itay4) [inline Packs/Trello/Integrations/Trello/Trello.yml:427]:
```suggestion
  dockerimage: demisto/python3:3.9.4.18682
```

**Review by Itay4 — COMMENTED** [review]
looks good, we're almost done here :)
i've added some comments, most will be handled by me

File: Packs/Trello/Classifiers/classifier-Trello_-_Incoming_Mapper.json [hunk 29]
```diff
@@ -0,0 +1,65 @@
...
+				},
+				"Trello Created By": {
+					"complex": null,
+					"simple": "create_action.memberCreator.fullName"
+				},
+				"Trello Details": {
+					"complex": null,
+					"simple": "desc"
+				},
+				"Trello ID": {
```
Comment (Itay4) [inline Packs/Trello/Classifiers/classifier-Trello_-_Incoming_Mapper.json:33]: the Trello ID field exist in the mapper but it does not exist in the incident fields
should it be removed from the mapper or added to the PR?

**Review by Itay4 — APPROVED** [review]
Great work !

**Pull request merged**